File: behavioral/strategy/java/src/CreditCardPaymentStrategy.java
public class CreditCardPaymentStrategy implements PaymentStrategy {
  private String cardNumber;
  private String cvv;
  private String expiryDate;

  public CreditCardPaymentStrategy(String cardNumber, String cvv, String expiryDate) {
    this.cardNumber = cardNumber;
    this.cvv = cvv;
    this.expiryDate = expiryDate;
  }

  @Override
  public boolean pay(double amount) {
    System.out.println("Paying " + amount + " via Credit Card with number " + cardNumber);
    return true;
  }
}

File: behavioral/strategy/ruby/lib/credit_card_payment_strategy.rb
require_relative 'payment_strategy'

class CreditCardPaymentStrategy < PaymentStrategy
  attr_reader :card_number, :cvv, :expiry_date

  def initialize(card_number, cvv, expiry_date)
    @card_number = card_number
    @cvv = cvv
    @expiry_date = expiry_date
  end

  # Overrides PaymentStrategy#pay
  def pay(amount)
    puts "Paying #{amount} via Credit Card with number #{card_number}"
    true
  end
end
